tests/protocols/test_http.py

```diff
@@ -716,6 +728,34 @@ async def test_supported_upgrade_request(protocol_cls):
     assert b"HTTP/1.1 426 " in protocol.transport.buffer
 
 
+@pytest.mark.anyio
+@pytest.mark.parametrize("protocol_cls", HTTP_PROTOCOLS)
+async def test_ignored_ws_upgrade_request(protocol_cls):
+    app = Response("Hello, world", media_type="text/plain")
+
+    protocol = get_connected_protocol(
+        app, protocol_cls, ws="wsproto", ws_ignore_upgrade=True
+    )
+    protocol.data_received(UPGRADE_REQUEST)
+    await protocol.loop.run_one()
+    assert b"HTTP/1.1 200 OK" in protocol.transport.buffer
+    assert b"Hello, world" in protocol.transport.buffer
+
+
+@pytest.mark.anyio
+@pytest.mark.parametrize("protocol_cls", HTTP_PROTOCOLS)
+async def test_http2_upgrade_request(protocol_cls):
+    app = Response("Hello, world", media_type="text/plain")
+
+    protocol = get_connected_protocol(
+        app, protocol_cls, ws="wsproto", ws_ignore_upgrade=True
```

Comment:
I've been thinking about this, and maybe it will be better if we remove the ws-ignore-upgrade in this test. 🤔

Reply:
This is a note to myself.

Reply:
Hm, I think it would be best to do this test with ws-ignore-upgrade set to both True and False, as the test case should have no knowledge of the implementation and it should in both cases be ignored.
I can make that change if you agree with me

Reply:
yes

Reply:
done

Reply:
thanks